### PR TITLE
GameBoard Refactoring

### PR DESCRIPTION
**Pull request by asiderop**
This PR splits the `JanggiGame` class into separate `Game` and `Board` classes; the associated unit test modules are also split. The `Board` class handles all lower-level grid manipulations (e.g. moves) and checks (e.g. out-of-bounds) and the `Game` class handles higher-level game logic (e.g. player turns) and rules. The class usage (where `->` means "has") is now: `JanggiGui -> Game -> Board -> Piece`

The `Board`'s internal 2d array is renamed from `_board` to `_grid` for clarity, and access from outside the `Board` class is removed.

Static utility functions (e.g. `numeric_to_algebraic()`) are moved into a new `jangii.utils` module, with unit tests moved as needed. The parameters for each function are annotated for static analysis.

The `Piece` class (and its children) are updated to annotate parameter types (helps IDE static analyzers) and makes use of the various functions that are moved (e.g. board checkers moving into `Board`, util routines moving into `utils.py`).

Lastly, there is some refactoring in the `gui.py` module to reduce duplicate code, and some overall PEP-style refactoring throughout the project.

### CODE REVIEW

**Comment by asiderop** [comment]
Okay. This is ready for review and merge, now. Thanks.

File: janggi/gui.py [hunk 1]
```diff
@@ -76,20 +76,32 @@ def get_board_rectangles():
...
     for a_string in list_of_strings:
         black = (0, 0, 0)
         img = font.render(a_string.upper(), True, black)
-        image_dict[a_string] = img
-    return image_dict
+        image_list.append(img)
+    return image_list
+
+
+def blit_images(screen, images: list[pygame.Surface], coord_x: int, coord_y: int, stride_x=0, stride_y=0):
```
Comment (zacharybmeyers) [inline janggi/gui.py:94]: This is king. Nice!!

File: janggi/game.py [hunk 2]
```diff
@@ -181,16 +230,12 @@ def make_move(self, start, end):
...
         # otherwise, VALID MOVE
-        start_tup = algebraic_to_numeric(start)
-        start_row, start_col = start_tup
-        end_row, end_col = end_tup  # unpack end square tup
-        board = self._board.get_board()  # get board
 
         # move Piece object to new square
         # (removes opposing piece or fills empty square)
-        board[end_row][end_col] = piece_obj
+        self._board.set_square_contents(end, piece_obj)
```
Comment (zacharybmeyers) [inline janggi/game.py:221]: Good catch!

**Comment by zacharybmeyers** [comment]
Smashed it. Thank you!! You did so much dirty work...

**Pull request merged**